src/Arcus.Security.Tests.Integration/KeyVault/SecretStoreBuilderExtensionsTests.cs

```diff
@@ -613,6 +614,42 @@ public async Task AddAzureKeyVaultWithTenant_WithServicePrincipal_GetSecretSucce
             Assert.NotNull(secret.Version);
         }
 
+        [Fact]
+        public async Task AddAzureKeyVaultWithTenant_WithServicePrincipal_GetSecretSucceeds()
```

Comment:
I could be using an testing approach were we use functions as first class citizens, but that would only make it easier for seasoned devs, not for everyone, so for now this is probably the best way.